expression/aggregation/descriptor.go

```diff
@@ -297,11 +297,16 @@ func (a *AggFuncDesc) typeInfer4Sum(ctx sessionctx.Context) {
 		if a.RetTp.Decimal < 0 || a.RetTp.Decimal > mysql.MaxDecimalScale {
 			a.RetTp.Decimal = mysql.MaxDecimalScale
 		}
+		a.RetTp.Flag |= a.Args[0].GetType().Flag & mysql.UnsignedFlag
 		// TODO: a.Args[0] = expression.WrapWithCastAsDecimal(ctx, a.Args[0])
-	default:
+	case mysql.TypeDouble, mysql.TypeFloat:
 		a.RetTp = types.NewFieldType(mysql.TypeDouble)
 		a.RetTp.Flen, a.RetTp.Decimal = mysql.MaxRealWidth, a.Args[0].GetType().Decimal
```

Comment:
should we also maintain the unsigned flag for double result?

Reply:
emm.. Seems that we just ignored the unsigned flag of double. Golang also don't have unsigned float64 too...